### PR TITLE
CLDR-14284 TestAttributeValues check for duplicates in space-separated attributes

### PR DESCRIPTION
**Pull request by srl295**
CLDR-14284 

- also removed a comment on a previously-duplicate grammaticalFeatures element
that was fixed with CLDR-13730

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-_____
- [ ] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestAttributeValues.java [hunk 1]
```diff
@@ -261,6 +262,14 @@ public void checkElement(String element, Attributes atts) {
         }
 
         private void checkAttribute(String element, String attribute, String attrValue) {
+            if(LdmlConvertRules.isSplittableAttr(element, attribute)) {
+                final Set<String> asSet = new HashSet<>();
+                SPACE_SPLITTER.splitToList(attrValue).forEach(v -> {
+                    if(asSet.add(v) == false) {
+                        throw new IllegalArgumentException("in <" + element+ " " + attribute + "=…, value " + v + " appears twice.");
```
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestAttributeValues.java:269]: This does print an error and stop the test. There may be a more graceful way to fail.

**Review by macchiati — CHANGES_REQUESTED** [review]
This wasn't using what I was thinking of. Will follow up with details

**Comment by srl295** [comment]
@macchiati let me know if this looks correct

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java [hunk 2]
```diff
@@ -616,7 +631,14 @@ public static SetMatchValue of(String key) {
 
         @Override
         public  boolean is(String items) {
-            return and(subtest,SPACE_SPLITTER.split(items));
+            Iterable<String> splitItems = SPACE_SPLITTER.split(items);
```
Comment (macchiati) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java:634]: Looks good. Should add quick unit test.
Also, can be pithier. Split to list, then create set from it. If the sizes are not the same, fail.

**Review by macchiati — CHANGES_REQUESTED** [review]

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java (hunk 2 above)
Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java:634]: > Should add quick unit test.

TestAttributeValues already hits this, and is verified to fail on the `"… fil fil …"` case for grammatical features.

> Also, can be pithier… If the sizes are not the same, fail

good point, will fix

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java [hunk 3]
```diff
@@ -640,7 +659,6 @@ public String getName() {
         }
 
         public static OrMatchValue of(String key) {
-            IntFunction<MatchValue[]> generator = null;
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java:643]: just a line of dead code, no side effect

**Review by macchiati — APPROVED** [review]

**Pull request merged**